homework/index.js

```diff
@@ -29,22 +29,74 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function createTag(name, options = {}) {
+    const elem = document.createElement(name);
+    Object.entries(options).forEach(([key, value]) => {
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  function buildTableRow(title, parent, content) {
+    const row = createAndAppend(`tr`, parent);
+    createAndAppend(`th`, row, { text: `${title}:` });
+    let data;
+    if (typeof content === `string` || typeof content === `number`) {
+      createAndAppend(`td`, row, { text: `${content}` });
+    } else {
+      data = createAndAppend(`td`, row);
+      data.appendChild(content);
+    }
+  }
+
+  function renderRepoDetails(repo, parent) {
+    const detailsDiv = createAndAppend(`div`, parent, { class: `details` });
+    const table = createAndAppend(`table`, detailsDiv);
+
+    const repoLink = createTag(`a`, {
+      href: repo.html_url,
+      target: '_blank',
+      text: repo.name,
+    });
+    buildTableRow(`Repository`, table, repoLink);
+
+    const description = repo.description
+      ? { text: repo.description }
+      : { text: 'No Description', class: 'alert-no-description' };
+
+    const repoDescription = createTag(`span`, description);
+    buildTableRow(`Description`, table, repoDescription);
```

Comment:
To deal with the Description you could use a so-called _short-circuit_ expression:

```js
buildTableRow(`Description`, table, repo.description || 'No Description');
```

Of course, this doesn't use the extra CSS class you were adding in the case of a missing description. If you would still like to add that, you can do the same trick as we did for the first row:

```js
const descriptionRow = buildTableRow(`Description`, table, '');
const description = repo.description
  ? { text: repo.description }
  : { text: 'No Description', class: 'alert-no-description' };
createAndAppend('span', descriptionRow.lastChild, description);
```

Note that we no longer need your `createTag` function, which looked like a stripped-down version of `createAndAppend`.